packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/StoreKit2/InAppPurchaseStoreKit2.swift

```diff
@@ -34,4 +34,150 @@ extension InAppPurchasePlugin: InAppPurchase2API {
+  func transactions(
+    completion: @escaping (Result<[SK2TransactionMessage], Error>) -> Void
+  ) {
+    Task {
+      @MainActor in
+      do {
+        let transactionsMsgs = await rawTransactions().map {
+          $0.convertToPigeon()
+        }
+        completion(.success(transactionsMsgs))
+      }
+    }
+  }
+
+  /// Wrapper method around StoreKit2's finish() method https://developer.apple.com/documentation/storekit/transaction/3749694-finish
+  func finish(id: Int64, completion: @escaping (Result<Void, Error>) -> Void) {
+    Task {
+      let transaction = try await fetchTransaction(by: UInt64(id))
+      if let transaction = transaction {
+        await transaction.finish()
+      }
+    }
+  }
+
+  /// This Task listens  to Transation.updates as shown here
+  /// https://developer.apple.com/documentation/storekit/transaction/3851206-updates
+  /// This function should be called as soon as the app starts to avoid missing any Transactions done outside of the app.
+  func startListeningToTransactions() throws {
+    self.updateListenerTask = Task { [weak self] in
+      for await verificationResult in Transaction.updates {
+        switch verificationResult {
+        case .verified(let transaction):
+          self?.sendTransactionUpdate(transaction: transaction)
+        case .unverified:
+          break
+        }
+      }
+    }
+  }
+
+  /// Stop subscribing to Transaction.updates
+  func stopListeningToTransactions() throws {
+    getUpdateListenerTask().cancel()
+  }
+
+  /// Sends an transaction back to Dart. Access these transactions with `purchaseStream`
+  func sendTransactionUpdate(transaction: Transaction) {
+    let transactionMsg = transaction.convertToPigeon()
+    transactionDelegate?.onTransactionsUpdated(newTransaction: transactionMsg) { result in
+      switch result {
+      case .success: break
+      case .failure(let error):
+        print("Failed to send transaction updates: \(error)")
+      }
+    }
+  }
+
+  /// Helper function that fetches and unwraps all verified transactions
+  private func rawTransactions() async -> [Transaction] {
+    var transactions: [Transaction] = []
+    for await verificationResult in Transaction.all {
+      switch verificationResult {
+      case .verified(let transaction):
+        transactions.append(transaction)
+      case .unverified:
+        break
+      }
+    }
+    return transactions
+  }
+
+  /// Helper function to fetch specific transaction
+  private func fetchTransaction(by id: UInt64) async throws -> Transaction? {
+    for await result in Transaction.all {
+      switch result {
+      case .verified(let transaction):
+        if transaction.id == id {
+          return transaction
+        }
+      case .unverified:
+        continue
+      }
+    }
+    return nil
+  }
+
+  /// Helper function to cast updateListenerTask to a task
+  func getUpdateListenerTask() -> Task<(), Never> {
```

Comment:
can you use a getter (rather than a function) in the class definition (rather than extension). And make `updateListenerTask` private.